tests/end-to-end/backup-restore-test/README.md

```diff
@@ -0,0 +1,122 @@
+5. Remove the Namespace and all its resources.
+6. Restore the Namespace and its resources from the backup.
+7. Verify if the resources contained in the restored Namespace work.
+
+### Use environment variables
+
+Use the following environment variables to configure the tests:
+
+| Name | Required | Default | Description |
+|-----|:---------:|:--------:|------------|
+| **DOMAIN** | NO | `kyma.local` | The domain where Kyma runs. |
+| **USER_EMAIL** | YES | - | The email address for authentication in Dex. |
+| **USER_PASSWORD** | YES | - | The password for authentication in Dex. |
+| **KUBECONFIG** | NO | - | The path to the `kubeconfig` file needed to run tests outside the cluster. |
+| **ALL_BACKUP_CONFIGURATION_FILE** | NO | `/all-backup.yaml` | The path to the `all-backup` configuration file. |
+| **SYSTEM_BACKUP_CONFIGURATION_FILE** | NO | `/system-backup.yaml` | The path to the `system-backup` configuration file. |
+
+
+## Development
+
+This section presents how to add and run a new test. 
+
+### Add a new test
+
+Add a new test under the `backupe2e/{domain-name}` directory and implement the following interface:
+
+```go
+type BackupTest interface {
+    CreateResources(namespace string)
+    TestResources(namespace string)
+    DeleteResources()
+}
+```
+The functions work as follows:
+
+- The `TestResources` function validates if the test data works as expected. 
+- The `CreateResources` function installs the required test data before the backup process starts.
+- The `DeleteResources` function deletes the resources that are a part of the cluster before executing the test. The resources need to be deleted to test the restore process.
+
+After the pipeline executes the backup and restore processes on the cluster, the `TestResources` function validates if the restore worked as expected.
+
+The test creates a new Namespace called `restore-test-{UUID}`. This Namespace contains all resources created during the test.
+
+### Run end-to-end tests locally
+
+> **NOTE:** Before running the test, configure Ark using [these](../../../docs/backup/docs/03-01-backup-configuration.md) guidelines.
+
+Run tests:
+```bash
+env KUBECONFIG={KUBECONFIG} go test ./... -count=1 -timeout=0
+```
+where `{KUBECONFIG}` is the path to the `kubeconfig` file.
+
+### Run tests using a Helm chart
+
+Run the tests using Helm:
+
+1. Prepare the data:
+
+```bash
+helm install deploy/chart/backup-test --namespace end-to-end --name backup-test
+```
+2. Run tests:
+
+```bash
+helm test backup-test
+```
+
+### Run tests using Telepresence
+[Telepresence](https://www.telepresence.io/) allows you to run tests locally while connecting a service to a remote Kubernetes cluster. It is helpful when the test needs access to other services in a cluster.
+
+1. [Install Telepresence](https://www.telepresence.io/reference/install).
+2. Run tests:
+```bash
+env KUBECONFIG={KUBECONFIG} telepresence --run go test ./... -count=1 -timeout=0
```

Comment:
``` bash
env KUBECONFIG={KUBECONFIG} ALL_BACKUP_CONFIGURATION_FILE={ALL_BACKUP_PATH} SYSTEM_BACKUP_CONFIGURATION_FILE={SYSTEM_BACKUP_PATH} telepresence --run go test ./... -count=1 -timeout=0
```
where:
- `{KUBECONFIG}` is the path to the `kubeconfig` file.
- `{ALL_BACKUP_PATH}` is the path to the `all-backup.yaml` file.
- `{SYSTEM_BACKUP_PATH}` is the path to the `system-backup.yaml` file.

Reply:
👍